pkg/cmd/get/get_crd_counts.go

```diff
@@ -0,0 +1,165 @@
+package get
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/jenkins-x/jx/pkg/cmd/helper"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/templates"
+)
+
+// CRDCountOptions the command line options
+type CRDCountOptions struct {
+	*opts.CommonOptions
+}
+
+type tableLine struct {
+	name      string
+	version   string
+	count     int
+	namespace string
+}
+
+var (
+	getCrdCountLong = templates.LongDesc(`
+		Count the number of resources for all custom resources definitions
+
+`)
+
+	getCrdCountExample = templates.Examples(`
+
+		# Count the number of resources for all custom resources definitions
+		jx get crd count
+	`)
+)
+
+// NewCmdGetCRDCount creates the command object
+func NewCmdGetCRDCount(commonOpts *opts.CommonOptions) *cobra.Command {
+	options := &CRDCountOptions{
+		CommonOptions: commonOpts,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "crd count",
+		Short:   "Display resources count for all custom resources",
+		Long:    getCrdCountLong,
+		Example: getCrdCountExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.Cmd = cmd
+			options.Args = args
+			err := options.Run()
+			helper.CheckErr(err)
+		},
+	}
+	return cmd
+}
+
+// Run implements this command
+func (o *CRDCountOptions) Run() error {
+	results, err := o.getCustomResourceCounts()
+	if err != nil {
+		return errors.Wrap(err, "cannot get custom resource counts")
+	}
+
+	table := o.CreateTable()
+	table.AddRow("NAME", "VERSION", "COUNT", "NAMESPACE")
+
+	for _, r := range results {
+		table.AddRow(r.name, r.version, strconv.Itoa(r.count), r.namespace)
+	}
+
+	table.Render()
+	return nil
+}
+
+func (o *CRDCountOptions) getCustomResourceCounts() ([]tableLine, error) {
+
+	exClient, err := o.ApiExtensionsClient()
```

Comment:
What's about extracting this into a package (crd?) outside of cmd structure, making it potentially reusable from other places. It could return a slice of CRDCount or similar which you then render into a table?

Reply:
Feels ok here to me until there's a time for reuse TBH.  If it concerns you a lot then sure I can do but I prefer not to unless there's real cases of reuse, I dont see any yet mind.